src/game/Pet.cpp

```diff
@@ -143,15 +143,6 @@ bool Pet::LoadPetFromDB(Player* owner, uint32 petentry, uint32 petnumber, bool c
     uint32 summon_spell_id = fields[21].GetUInt32();
     SpellEntry const* spellInfo = sSpellStore.LookupEntry(summon_spell_id);
 
-    bool is_temporary_summoned = spellInfo && GetSpellDuration(spellInfo) > 0;
```

Comment:
Yes there is; but he is not a summon, he is a mind-control; you don't get a mind-control as your pet, do you? I know you get it as your controlled, but you get to actually save your mind control in the db and stuff like that? Cause if a player disconnects while mind-controlling another player, then how do we handle that?

Picture this; one guy comes to mind control the dragon for the quest, he disconnects. Then another guy whos up to do the quest, he mind-controls the dragon. Now, while he is still mind-controlling the dragon, the first guy reconnects. How do we handle this?

I say there is one uniform sollution to all of this; break mind-control on disconnect. I see no other sollution without entering some weird very unwanted scenarios.